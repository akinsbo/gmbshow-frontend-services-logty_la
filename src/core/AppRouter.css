body {
    background: blue;
}


/* @font-face {
    font-family: 'MyFont';
    src: url('../fonts/Mosk\ Typeface/Mosk\ Normal\ 400.ttf') format('ttf'), url('../fonts/Mosk\ Typeface/Mosk\ Light\ 300.ttf') format('ttf');
    font-weight: 600;
    font-style: normal;
} */

.hello {
    color: red;
    /* font-family: 'MyFont'; */
    background: url('../images/astronaut.png');
}