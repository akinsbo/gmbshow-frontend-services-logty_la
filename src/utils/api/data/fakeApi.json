{
	"user": [
		{
			"name": "elit dolore amet sunt deserunt",
			"email": "commodo ea tempor laborum"
		},
		{
			"name": "ea cupidatat",
			"email": "occaecat ea velit"
		},
		{
			"name": "ut ut Lorem magna nostrud",
			"email": "amet"
		},
		{
			"name": "dolore elit in",
			"email": "aliquip amet id consectetur"
		},
		{
			"name": "Ut aute aliquip",
			"email": "pariatur voluptate reprehenderit ipsum"
		},
		{
			"name": "non officia",
			"email": "consectetur non in Lorem"
		},
		{
			"name": "in Excepteur reprehenderit",
			"email": "enim"
		},
		{
			"name": "Excepteur incididunt exercitation pariatur",
			"email": "velit et commodo"
		},
		{
			"name": "Duis Excepteur sit",
			"email": "in cillum et irure qui"
		},
		{
			"name": "laboris officia commodo pariatur consectetur",
			"email": "culpa ut"
		},
		{
			"name": "elit ipsum",
			"email": "aliqua voluptate ipsum mollit"
		},
		{
			"name": "fugiat",
			"email": "commodo cupidatat nostrud"
		},
		{
			"name": "ut consectetur irure",
			"email": "sunt proident laboris Excepteur"
		},
		{
			"name": "voluptate ut quis nostrud",
			"email": "officia ex"
		},
		{
			"name": "sed ea in",
			"email": "cupidatat Ut anim proident ullamco"
		},
		{
			"name": "in irure",
			"email": "nulla"
		},
		{
			"name": "quis id elit exercitation reprehenderit",
			"email": "elit pariatur aliquip Duis commodo"
		},
		{
			"name": "Excepteur",
			"email": "sint sunt esse"
		},
		{
			"name": "esse eiusmod ut",
			"email": "dolore ullamco eiusmod culpa"
		},
		{
			"name": "nostrud in esse dolor",
			"email": "elit sunt"
		},
		{
			"name": "dolore nisi et sed",
			"email": "veniam reprehenderit"
		},
		{
			"name": "ullamco consequat reprehenderit eu ad",
			"email": "laborum Lorem ut cupidatat consectetur"
		},
		{
			"name": "Ut velit",
			"email": "in nulla"
		},
		{
			"name": "consectetur ea laborum nulla qui",
			"email": "officia elit"
		},
		{
			"name": "in",
			"email": "tempor officia proident anim et"
		},
		{
			"name": "quis commodo ad",
			"email": "incididunt"
		},
		{
			"name": "ea veniam Duis",
			"email": "Lorem pariatur ad amet"
		},
		{
			"name": "eiusmod sit nisi Ut",
			"email": "ut quis reprehenderit"
		},
		{
			"name": "laborum ad consequat",
			"email": "Ut esse id"
		},
		{
			"name": "mollit Excepteur pariatur officia",
			"email": "quis elit"
		},
		{
			"name": "anim consequat do",
			"email": "Lorem exercitation ea"
		},
		{
			"name": "est incididunt Excepteur reprehenderit",
			"email": "adipisicing et est sunt"
		},
		{
			"name": "Lorem in",
			"email": "in in"
		},
		{
			"name": "consequat in sunt aliqua",
			"email": "nulla Lorem culpa ad non"
		},
		{
			"name": "velit cupidatat sed irure pariatur",
			"email": "laboris"
		},
		{
			"name": "magna eiusmod enim",
			"email": "ut in culpa in"
		},
		{
			"name": "eu aute ex",
			"email": "aute sunt adipisicing dolore"
		},
		{
			"name": "labore irure",
			"email": "amet quis cillum nulla"
		},
		{
			"name": "adipisicing aliqua ea",
			"email": "Ut id occaecat veniam ea"
		},
		{
			"name": "officia ad aliqua amet",
			"email": "adipisicing velit"
		},
		{
			"name": "elit",
			"email": "elit sint cupidatat"
		},
		{
			"name": "et aute reprehenderit",
			"email": "esse pariatur sit ex dolore"
		},
		{
			"name": "occaecat in in ex dolor",
			"email": "ipsum voluptate dolor cillum dolore"
		},
		{
			"name": "culpa in labore",
			"email": "dolore laboris"
		},
		{
			"name": "ea cillum occaecat nisi Lorem",
			"email": "velit commodo tempor ipsum"
		},
		{
			"name": "amet exercitation nostrud esse Duis",
			"email": "amet sint mollit"
		},
		{
			"name": "minim",
			"email": "labore quis aute proident"
		},
		{
			"name": "laboris et aute sit ut",
			"email": "culpa"
		},
		{
			"name": "in id voluptate",
			"email": "aliquip nostrud Ut"
		},
		{
			"name": "sit Lorem dolor",
			"email": "proident consequat aute"
		},
		{
			"name": "Excepteur sit in irure",
			"email": "nulla dolore"
		},
		{
			"name": "fugiat",
			"email": "veniam aute labore"
		},
		{
			"name": "esse aliquip eiusmod dolor",
			"email": "pariatur ea est"
		},
		{
			"name": "dolor voluptate amet elit officia",
			"email": "consequat ipsum tempor officia pariatur"
		},
		{
			"name": "nostrud",
			"email": "in"
		},
		{
			"name": "dolor amet dolor ipsum",
			"email": "aute nulla"
		},
		{
			"name": "Lorem laboris quis pariatur magna",
			"email": "Excepteur"
		},
		{
			"name": "deserunt eiusmod tempor Excepteur",
			"email": "labore ad"
		},
		{
			"name": "sit ipsum",
			"email": "minim Ut reprehenderit et"
		},
		{
			"name": "dolor magna nulla eu incididunt",
			"email": "irure ut nostrud"
		},
		{
			"name": "enim sit",
			"email": "esse"
		},
		{
			"name": "magna",
			"email": "eiusmod"
		},
		{
			"name": "officia laboris tempor",
			"email": "deserunt Lorem"
		},
		{
			"name": "ut sit nulla in dolor",
			"email": "eiusmod aute"
		},
		{
			"name": "nisi id dolore aliqua ad",
			"email": "irure nisi"
		},
		{
			"name": "exercitation eu quis",
			"email": "sed Excepteur"
		},
		{
			"name": "ullamco",
			"email": "in veniam aliquip amet"
		},
		{
			"name": "adipisicing qui",
			"email": "in veniam in aliquip"
		},
		{
			"name": "reprehenderit",
			"email": "sed ex culpa commodo voluptate"
		},
		{
			"name": "do culpa",
			"email": "exercitation"
		},
		{
			"name": "eu id amet quis dolor",
			"email": "esse elit culpa reprehenderit sed"
		},
		{
			"name": "ut eu",
			"email": "ut anim"
		},
		{
			"name": "proident sit id in occaecat",
			"email": "laborum esse reprehenderit"
		},
		{
			"name": "ut veniam amet",
			"email": "cillum exercitation"
		},
		{
			"name": "nostrud dolor mollit",
			"email": "ea Ut"
		},
		{
			"name": "minim",
			"email": "enim esse irure"
		},
		{
			"name": "labore",
			"email": "amet labore laborum dolor"
		},
		{
			"name": "occaecat dolor sed",
			"email": "quis"
		},
		{
			"name": "pariatur",
			"email": "dolore"
		},
		{
			"name": "sunt enim sit Lorem irure",
			"email": "ad"
		},
		{
			"name": "anim",
			"email": "tempor Duis qui"
		},
		{
			"name": "veniam amet tempor occaecat sit",
			"email": "aute cupidatat consectetur sint officia"
		},
		{
			"name": "nostrud",
			"email": "nostrud adipisicing irure aute commodo"
		},
		{
			"name": "non",
			"email": "deserunt cillum ullamco"
		},
		{
			"name": "ut id",
			"email": "esse"
		},
		{
			"name": "mollit in",
			"email": "sunt ullamco dolor in exercitation"
		},
		{
			"name": "est id aliqua",
			"email": "magna dolor"
		},
		{
			"name": "voluptate eiusmod exercitation elit in",
			"email": "Excepteur Duis amet"
		},
		{
			"name": "sit",
			"email": "nulla ad"
		},
		{
			"name": "ullamco magna sit",
			"email": "tempor irure culpa"
		},
		{
			"name": "in dolor qui",
			"email": "ea proident consequat consectetur"
		},
		{
			"name": "aliquip elit nostrud dolore ipsum",
			"email": "ea est voluptate in ipsum"
		},
		{
			"name": "fugiat nostrud voluptate ullamco",
			"email": "consectetur culpa qui"
		},
		{
			"name": "aute",
			"email": "labore non culpa enim Excepteur"
		},
		{
			"name": "Excepteur labore enim ad commodo",
			"email": "amet Lorem sint magna"
		},
		{
			"name": "dolore",
			"email": "eiusmod fugiat"
		},
		{
			"name": "dolore",
			"email": "voluptate"
		},
		{
			"name": "minim reprehenderit officia sed",
			"email": "voluptate irure laborum magna id"
		},
		{
			"name": "in laboris esse Excepteur pariatur",
			"email": "sunt in minim"
		},
		{
			"name": "eiusmod",
			"email": "dolor commodo sit"
		},
		{
			"name": "in Lorem fugiat Duis",
			"email": "eiusmod amet pariatur ea"
		},
		{
			"name": "ad consectetur",
			"email": "sit velit magna ut"
		},
		{
			"name": "et",
			"email": "aute Ut adipisicing ipsum"
		},
		{
			"name": "in ullamco culpa in",
			"email": "quis tempor ut"
		},
		{
			"name": "aute aliqua ut laborum consequat",
			"email": "ea fugiat sint commodo ullamco"
		},
		{
			"name": "et aute",
			"email": "cillum dolor ut"
		}
	]
}