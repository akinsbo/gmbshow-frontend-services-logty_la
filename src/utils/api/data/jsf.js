const jsf = require('json-schema-faker')
const schema = require('./mockDataSchema')
const fs = require('fs-extra')
const chalk = require('chalk')
const log = console.log

/**
 * This file gxenerates the JSON file to be served by Json-server
 */
let apiJson
const output_file_path = "fakeApi.json"

jsf.resolve(schema)
    .then(schemObjectData => {
        apiJson = JSON.stringify(schemObjectData, null, '\t')
        log(chalk.blue(`apiJson = ${apiJson}`))

        // Write into file
        fs.outputFile(output_file_path, apiJson)
            .then(data => {
                log(chalk.green("Mock data generated."))
            })
            .catch(error => {
                log(chalk.red(`output file error ${error}`))
            })
    }).catch(error => log(chalk.red(`schema is not resolved: ${error}`)))