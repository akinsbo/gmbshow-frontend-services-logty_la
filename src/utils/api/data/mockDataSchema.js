module.exports = {
    "type": "object",
    "properties": {
        "user": {
            "type": "array",
            "minItems": 100,
            "maxItems": 200,
            "items": {
                "type": "object",
                "properties": {
                    "name": {
                        "type": "string",
                        "faker": "name.findName"
                    },
                    "email": {
                        "type": "string",
                        "faker": "internet.email"
                    }
                },
                "required": [
                    "name",
                    "email"
                ]
            }
        }
    }
}