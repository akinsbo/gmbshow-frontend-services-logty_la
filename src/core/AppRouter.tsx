import * as React from "react"
import { Link, Route, Switch } from "react-router-dom"
import HomePage from "../pages/HomePage"
import SecondPage from "../pages/SecondPage"
import ThirdPage from "../pages/ThirdPage"
import * as styles from "./AppRouter.css"
import HeadModifier from "./HeadModifier"

// 'AppRouterProps' describes the shape of props.
// State is never set so we use the '{}' type.
export default class AppRouter extends React.Component<{}, {}> {
  render() {
    return (
      // Links would be in another file
      <div className={styles.hello}>
        <HeadModifier />
        <nav>
          <Link to="/">HomePage</Link>
          <Link to="/page2">SecondPage</Link>
          <Link to="/page3">ThirdPage</Link>
        </nav>
        <div>
          <h1>My Custom React app</h1>
          <Switch>
            <Route exact={true} path="/" component={HomePage} />
            <Route path="/page2" component={SecondPage} />
            <Route path="/page3" component={ThirdPage} />
          </Switch>
        </div>
      </div>
    )
  }
}
