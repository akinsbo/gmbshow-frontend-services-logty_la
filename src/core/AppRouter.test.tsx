import * as React from "react"
import * as renderer from "react-test-renderer"
import { BrowserRouter as Router } from "react-router-dom"
import data from "../site-data.json"
import { expect } from "chai"

import AppRouter from "./AppRouter"

describe("<AppRouter />", () => {
  const wrapper = renderer.create(
    <Router>
      <AppRouter siteData={data} />
    </Router>
  )

  test("render", () => {
    expect(wrapper).toMatchSnapshot()
  })

  describe("structure", () => {
    describe("Helmet", () => {
      let helmetWrapper
      beforeEach(() => {
        helmetWrapper = wrapper.find("Helmet")
      })
      it("should be present", () => {
        expect(helmetWrapper).toHaveLength(1)
      })
      it("should contain elements", () => {
        expect(helmetWrapper)
          .find("title")
          .toHaveLength(1)
        describe("meta", () => {
          const metaWrapper = wrapper.find("meta")
          it("should contain meta element", () => {
            expect(metaWrapper).toHaveLength(1)
          })
          it("should contain props", () => {
            expect(metaWrapper).toHaveProp("name")
          })
        })
      })
    })
  })
})
