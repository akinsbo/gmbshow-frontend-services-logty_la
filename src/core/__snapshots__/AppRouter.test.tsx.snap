// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppRouter should render 1`] = `
<div>
  <div />
  <nav>
    <a
      href="/"
      onClick={[Function]}
    >
      HomePage
    </a>
    <a
      href="/page2"
      onClick={[Function]}
    >
      SecondPage
    </a>
    <a
      href="/page3"
      onClick={[Function]}
    >
      ThirdPage
    </a>
  </nav>
  <div
    className="makeStyles-grow-2"
  >
    <div
      className="application"
    >
      <div />
    </div>
    <h1>
      This is the Home page
    </h1>
    <img
      alt="duyonoStonePng"
      src="test-file-stub"
    />
  </div>
  <div>
    <h1>
      My Custom React app
    </h1>
  </div>
</div>
`;
